File: cpp/0169_majorityElement.cpp
// 1.Hash, 但空间占O(n)
// 2.Vote： 维护一个count，实时更新众数，因为众数一定会剩下，所以count>0的数即为众数
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int res;
        for(int num:nums)
        {
            if(count==0)
                res = num;
            count += res==num? 1 : -1;
        }
        return res;
    }
};

File: java/0169_majorityElement.java
class Solution {
    public int majorityElement(int[] nums) {
        int count=1;
        int res=nums[0];
        for(var i=1;i<nums.length;i++)
        {
            if(count==0)
                res = nums[i];
            count += res==nums[i]? 1 : -1;
        }
        return res;
    }
}
